From 6c5c413afc634ddc4585e3dc9f8a3f27e00249d7 Mon Sep 17 00:00:00 2001
From: De Huo <De.Huo@windriver.com>
Date: Wed, 24 Jul 2019 18:04:46 +0800
Subject: [PATCH] Update rules accounts_passwords_pam_tally2_deny and
 accounts_passwords_pam_tally2_deny_root

Update the content in rules accounts_passwords_pam_tally2_deny and
accounts_passwords_pam_tally2_deny_root.

Upstream-Status: Pending

Signed-off-by: De Huo <De.Huo@windriver.com>
---
 .../bash/wrlinux1019.sh                            |  8 ++--
 .../accounts_passwords_pam_tally2_deny/rule.yml    |  2 +-
 .../ansible/shared.yml                             | 32 ++++++-------
 .../rule.yml                                       | 54 ++++++++++------------
 4 files changed, 44 insertions(+), 52 deletions(-)

diff --git a/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny/bash/wrlinux1019.sh b/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny/bash/wrlinux1019.sh
index 6a51638..d96a7fc 100644
--- a/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny/bash/wrlinux1019.sh
+++ b/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny/bash/wrlinux1019.sh
@@ -1,16 +1,16 @@
 # platform = multi_platform_wrlinux
 
 . /usr/share/scap-security-guide/remediation_functions
-populate var_accounts_passwords_pam_faillock_deny
+populate var_accounts_passwords_pam_tally2_deny
 
 if grep -q '^[^#].*pam_tally2.so' /etc/pam.d/common-auth; then
   if grep -q 'deny=' /etc/pam.d/common-auth; then
-	  sed -i "s/deny=[0-9]*/deny=$var_accounts_passwords_pam_faillock_deny/" /etc/pam.d/common-auth
+	  sed -i "s/deny=[0-9]*/deny=$var_accounts_passwords_pam_tally2_deny/" /etc/pam.d/common-auth
   else
-	  sed -i "/auth.*pam_tally2.so/s/$/ deny=$var_accounts_passwords_pam_faillock_deny/" /etc/pam.d/common-auth
+	  sed -i "/auth.*pam_tally2.so/s/$/ deny=$var_accounts_passwords_pam_tally2_deny/" /etc/pam.d/common-auth
   fi
 else
-  echo "auth  required pam_tally2.so deny=$var_accounts_passwords_pam_faillock_deny" >>/etc/pam.d/common-auth
+  echo "auth  required pam_tally2.so deny=$var_accounts_passwords_pam_tally2_deny" >>/etc/pam.d/common-auth
 fi
 
 echo "account	required	pam_tally2.so" >>/etc/pam.d/common-account
diff --git a/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny/rule.yml b/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny/rule.yml
index 29c18eb..4ebd097 100644
--- a/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny/rule.yml
+++ b/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny/rule.yml
@@ -11,7 +11,7 @@ description: |-
     <br /><br />
     <ul>
     <li> add the following line immediately <tt>before</tt> the <tt>pam_unix.so</tt> statement in the <tt>AUTH</tt> section:
-    <pre>auth required pam_tally2.so preauth silent deny=<sub idref="var_accounts_passwords_pam_tally2_deny" /> unlock_time=<sub idref="var_accounts_passwords_pam_tally2_deny" /> fail_interval=<sub idref="var_accounts_passwords_pam_faillock_fail_interval" /></pre></li>
+    <pre>auth required pam_tally2.so preauth silent deny=<sub idref="var_accounts_passwords_pam_tally2_deny" /> unlock_time=<sub idref="var_accounts_passwords_pam_faillock_unlock_time" /> fail_interval=<sub idref="var_accounts_passwords_pam_faillock_fail_interval" /></pre></li>
     <li> add the following line immediately <tt>after</tt> the <tt>pam_unix.so</tt> statement in the <tt>AUTH</tt> section:
     <pre>auth [default=die] pam_tally2.so authfail deny=<sub idref="var_accounts_passwords_pam_tally2_deny" /> unlock_time=<sub idref="var_accounts_passwords_pam_faillock_unlock_time" /> fail_interval=<sub idref="var_accounts_passwords_pam_faillock_fail_interval" /></pre></li>
     <li> add the following line immediately <tt>before</tt> the <tt>pam_unix.so</tt> statement in the <tt>ACCOUNT</tt> section:
diff --git a/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny_root/ansible/shared.yml b/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny_root/ansible/shared.yml
index d2b08c0..74be002 100644
--- a/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny_root/ansible/shared.yml
+++ b/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny_root/ansible/shared.yml
@@ -1,11 +1,10 @@
-# platform = multi_platform_rhel,multi_platform_fedora,multi_platform_ol,multi_platform_rhv
+# platform = multi_platform_wrlinux,multi_platform_rhel,multi_platform_fedora,multi_platform_ol,multi_platform_rhv
 # reboot = false
 # strategy = restrict
 # complexity = low
 # disruption = low
-- (xccdf-var var_accounts_passwords_pam_faillock_deny)
 
-- name: Add auth pam_faillock preauth deny before pam_unix.so
+- name: Add auth pam_tally2 preauth even_deny_root before pam_unix.so
   pamd:
     name: "{{ item }}"
     type: auth
@@ -13,30 +12,30 @@
     module_path: pam_unix.so
     new_type: auth
     new_control: required
-    new_module_path: pam_faillock.so
+    new_module_path: pam_tally2.so
     module_arguments: 'preauth
         silent
-        deny={{ var_accounts_passwords_pam_faillock_deny }}'
+        even_deny_root'
     state: before
   loop:
     - system-auth
     - password-auth
 
-- name: Add deny argument to auth pam_faillock preauth
+- name: Add even_deny_root argument to auth pam_tally2 preauth
   pamd:
     name: "{{ item }}"
     type: auth
     control: required
-    module_path: pam_faillock.so
+    module_path: pam_tally2.so
     module_arguments: 'preauth
         silent
-        deny={{ var_accounts_passwords_pam_faillock_deny }}'
+        even_deny_root'
     state: args_present
   loop:
     - system-auth
     - password-auth
 
-- name: Add auth pam_faillock authfail deny after pam_unix.so
+- name: Add auth pam_tally2 authfail even_deny_root after pam_unix.so
   pamd:
     name: "{{ item }}"
     type: auth
@@ -44,29 +43,28 @@
     module_path: pam_unix.so
     new_type: auth
     new_control: '[default=die]'
-    new_module_path: pam_faillock.so
+    new_module_path: pam_tally2.so
     module_arguments: 'authfail
-        deny={{ var_accounts_passwords_pam_faillock_deny }}'
+        even_deny_root'
     state: after
   loop:
     - system-auth
     - password-auth
 
-- name: Add deny argument to auth pam_faillock authfail
+- name: Add even_deny_root argument to auth pam_tally2 authfail
   pamd:
     name: "{{ item }}"
     type: auth
-    new_type: auth
     control: '[default=die]'
-    module_path: pam_faillock.so
+    module_path: pam_tally2.so
     module_arguments: 'authfail
-        deny={{ var_accounts_passwords_pam_faillock_deny }}'
+        even_deny_root'
     state: args_present
   loop:
     - system-auth
     - password-auth
 
-- name: Add account pam_faillock before pam_unix.so
+- name: Add account pam_tally2 before pam_unix.so
   pamd:
     name: "{{ item }}"
     type: account
@@ -74,7 +72,7 @@
     module_path: pam_unix.so
     new_type: account
     new_control: required
-    new_module_path: pam_faillock.so
+    new_module_path: pam_pam_tally2.so
     state: before
   loop:
     - system-auth
diff --git a/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny_root/rule.yml b/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny_root/rule.yml
index 29c18eb..be458d7 100644
--- a/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny_root/rule.yml
+++ b/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2_deny_root/rule.yml
@@ -1,49 +1,42 @@
 documentation_complete: true
 
-prodtype: wrlinux1019,rhel6,rhel7,rhel8,fedora,ol7,ol8,rhv4
+prodtype: wrlinux1019,rhel7,rhel8,fedora,ol7,ol8,rhv4
 
-title: 'Set Deny For Failed Password Attempts'
+title: 'Configure the root Account for Failed Password Attempts'
 
 description: |-
-    To configure the system to lock out accounts after a number of incorrect login
-    attempts using <tt>pam_tally2.so</tt>, modify the content of both
-    <tt>/etc/pam.d/common-auth</tt> and <tt>/etc/pam.d/common-account</tt> as follows:
+    To configure the system to lock out the <tt>root</tt> account after a
+    number of incorrect login attempts using <tt>pam_tally2.so</tt>, modify
+    the content of both <tt>/etc/pam.d/system-auth</tt> and
+    <tt>/etc/pam.d/password-auth</tt> as follows:
     <br /><br />
     <ul>
-    <li> add the following line immediately <tt>before</tt> the <tt>pam_unix.so</tt> statement in the <tt>AUTH</tt> section:
-    <pre>auth required pam_tally2.so preauth silent deny=<sub idref="var_accounts_passwords_pam_tally2_deny" /> unlock_time=<sub idref="var_accounts_passwords_pam_tally2_deny" /> fail_interval=<sub idref="var_accounts_passwords_pam_faillock_fail_interval" /></pre></li>
-    <li> add the following line immediately <tt>after</tt> the <tt>pam_unix.so</tt> statement in the <tt>AUTH</tt> section:
-    <pre>auth [default=die] pam_tally2.so authfail deny=<sub idref="var_accounts_passwords_pam_tally2_deny" /> unlock_time=<sub idref="var_accounts_passwords_pam_faillock_unlock_time" /> fail_interval=<sub idref="var_accounts_passwords_pam_faillock_fail_interval" /></pre></li>
-    <li> add the following line immediately <tt>before</tt> the <tt>pam_unix.so</tt> statement in the <tt>ACCOUNT</tt> section:
-    <pre>account required pam_tally2.so</pre></li>
+    <li>Modify the following line in the <tt>AUTH</tt> section to add
+    <tt>even_deny_root</tt>:
+    <pre>auth required pam_tally2.so preauth silent <b>even_deny_root</b> deny=<sub idref="var_accounts_passwords_pam_tally2_deny" /> unlock_time=<sub idref="var_accounts_passwords_pam_faillock_unlock_time" /> fail_interval=<sub idref="var_accounts_passwords_pam_faillock_fail_interval" /></pre></li>
+    <li>Modify the following line in the <tt>AUTH</tt> section to add
+    <tt>even_deny_root</tt>:
+    <pre>auth [default=die] pam_tally2.so authfail <b>even_deny_root</b> deny=<sub idref="var_accounts_passwords_pam_pam_tally2_deny" /> unlock_time=<sub idref="var_accounts_passwords_pam_faillock_unlock_time" /> fail_interval=<sub idref="var_accounts_passwords_pam_faillock_fail_interval" /></pre>
+    </li>
     </ul>
 
 rationale: |-
-    Locking out user accounts after a number of incorrect attempts
-    prevents direct password guessing attacks.
+    By limiting the number of failed logon attempts, the risk of unauthorized system access via user password
+    guessing, otherwise known as brute-forcing, is reduced. Limits are imposed by locking the account.
 
 severity: medium
 
 identifiers:
-    cce@rhel6: 26844-1
-    cce@rhel7: 27350-8
-    cce@rhel8: 80667-9
+    cce@rhel7: 80353-6
+    cce@rhel8: 80668-7
 
 references:
-    stigid@rhel6: RHEL-06-000061
-    srg@rhel6: SRG-OS-000021
-    disa@rhel6: '44'
-    cis: 5.3.2
-    cjis: 5.5.3
-    cui: 3.1.8
-    disa: "2238,44"
-    nist: AC-7(a)
+    disa: "2238"
+    nist: AC-7(b)
     nist-csf: PR.AC-7
     ospp: FMT_MOF_EXT.1
-    pcidss: Req-8.1.6
     srg: SRG-OS-000329-GPOS-00128,SRG-OS-000021-GPOS-00005
-    vmmsrg: SRG-OS-000021-VMM-000050
-    stigid@rhel7: "010320"
+    stigid@rhel7: "010330"
     isa-62443-2013: 'SR 1.1,SR 1.10,SR 1.2,SR 1.5,SR 1.7,SR 1.8,SR 1.9'
     isa-62443-2009: 4.3.3.6.1,4.3.3.6.2,4.3.3.6.3,4.3.3.6.4,4.3.3.6.5,4.3.3.6.6,4.3.3.6.7,4.3.3.6.8,4.3.3.6.9
     cobit5: DSS05.04,DSS05.10,DSS06.10
@@ -53,8 +46,9 @@ references:
 ocil_clause: 'that is not the case'
 
 ocil: |-
-    To ensure the failed password attempt policy is configured correctly, run the following command:
-    <pre>$ grep pam_tally2 /etc/pam.d/system-auth</pre>
-    The output should show <tt>deny=<sub idref="var_accounts_passwords_pam_tally2_deny" /></tt>.
+    To ensure that even the <tt>root</tt> account is locked after a defined number of failed password
+    attempts, run the following command:
+    <pre>$ grep even_deny_root /etc/pam.d/system-auth</pre>
+    The output should show <tt>even_deny_root</tt>.
 
 platform: pam
-- 
1.9.1

